MapboxNavigation/MapboxSpeechController.swift

```diff
@@ -0,0 +1,162 @@
+        
+        if let accessToken = accessToken, let host = host {
+            speech = SpeechSynthesizer(accessToken: accessToken, host: host)
+        } else if let accessToken = accessToken {
+            speech = SpeechSynthesizer(accessToken: accessToken)
+        } else {
+            speech = SpeechSynthesizer.shared
+        }
+        
+        spokenInstructionsForRoute.countLimit = 200
+        
+        super.init()
+    }
+    
+    @objc public override func didPassSpokenInstructionPoint(notification: NSNotification) {
+        let routeProgresss = notification.userInfo![MBRouteControllerDidPassSpokenInstructionPointRouteProgressKey] as! RouteProgress
+        for (stepIndex, step) in routeProgresss.currentLegProgress.leg.steps.suffix(from: routeProgresss.currentLegProgress.stepIndex).enumerated() {
+            let adjustedStepIndex = stepIndex + routeProgresss.currentLegProgress.stepIndex
+            
+            guard adjustedStepIndex < routeProgresss.currentLegProgress.stepIndex + stepsAheadToCache else { continue }
+            guard let instructions = step.instructionsSpokenAlongStep else { continue }
+            
+            for instruction in instructions {
+                guard spokenInstructionsForRoute.object(forKey: instruction.ssmlText as NSString) == nil else { continue }
+                
+                cacheSpokenInstruction(instruction: instruction)
+            }
+        }
+        
+        super.didPassSpokenInstructionPoint(notification: notification)
+    }
+    
+    public override func speak(_ instruction: SpokenInstruction) {
+        if let audioPlayer = audioPlayer, audioPlayer.isPlaying, let lastSpokenInstruction = lastSpokenInstruction {
+            voiceControllerDelegate?.voiceController?(self, didInterrupt: lastSpokenInstruction, with: instruction)
+        }
+        audioTask?.cancel()
+        audioPlayer?.stop()
+        lastSpokenInstruction = instruction
+        
+        guard spokenInstructionsForRoute.object(forKey: instruction.ssmlText as NSString) == nil else {
+            play(spokenInstructionsForRoute.object(forKey: instruction.ssmlText as NSString)! as Data)
+            return
+        }
+        
+        fetch(instruction: instruction)
+    }
+    
+    func speakWithoutPolly(_ instruction: SpokenInstruction, error: Error) {
+        audioTask?.cancel()
+        
+        voiceControllerDelegate?.voiceController?(self, spokenInstructionsDidFailWith: error)
+        
+        guard let audioPlayer = audioPlayer else {
+            super.speak(instruction)
+            return
+        }
+        
+        guard !audioPlayer.isPlaying else { return }
+        
+        super.speak(instruction)
+    }
+    
+    func fetch(instruction: SpokenInstruction) {
+        audioTask?.cancel()
+        let options = SpeechOptions(ssml: instruction.ssmlText)
+        if let locale = locale {
+            options.locale = locale
```

Comment:
We should figure out a way to let this code know the value of `RouteOptions.locale` so it can automatically match the language of the SSML coming from the Directions API.

Reply:
The good news is that the default locale value is calculated in the same manner across libs. But yeah, its almost like we need a public `LocaleManager` class.

Reply:
@1ec5 changed in https://github.com/mapbox/mapbox-navigation-ios/pull/617/commits/f12ea1b8ecb42c127098d9eea7647e233d9ee801. What are your thoughts on the new API?